src/lib/y2slp/clients/slp_server.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Slp
+  module Clients
+    # SlpServer main client
+    class SlpServer < Yast::Client
+      include Yast::Logger
+
+      # Constructor
+      def initialize
+        Yast.import "UI"
+        textdomain "slp-server"
+
+        Yast.import "Progress"
+        Yast.import "Report"
+        Yast.import "Summary"
+        Yast.import "CommandLine"
+        Yast.include self, "slp-server/wizards.rb"
+      end
+
+      def main
+        log_and_return do
+          propose? ? SlpServerAutoSequence() : CommandLine.Run(cmdline_description)
+        end
+      end
+
+    private
+
+      # It logs the start and finish of the given block call returning the
+      # result of the call.
```

Comment:
It says, "returning the result of the call", but `ret` is not the returned value. Am I wrong?

Reply:
You are completely right, fixing it.